src/main/java/com/newrelic/opentracing/aws/HeadersParser.java

```diff
@@ -30,10 +30,19 @@ private HeadersParser() {}
     try {
       if (input instanceof Map) {
         Map map = (Map) input;
-        final Object headers = map.get("headers");
-        if (headers instanceof Map) {
-          final Map<String, String> headerStr = (Map<String, String>) headers;
-          return tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(headerStr));
+        /*
+         * New Relic distributed tracing headers can be directly passed as Input to an AWS Lambda function
+         * as a Map with "newrelic" as the key and the distributed tracing payload String as the value or as
+         * a Map of headers.
+         */
+        if (map.get("newrelic") != null) {
+          return tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(map));
+        } else {
+          final Object headers = map.get("headers");
+          if (headers instanceof Map) {
+            final Map<String, String> headerStr = (Map<String, String>) headers;
+            return tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMapExtractAdapter(headerStr));
+          }
```

Comment:
do we have unit tests for this functionality? If not, let's make 'em.

Reply:
Not yet, looking into testing currently.